automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/ClusterConcurrencySpec.scala

```diff
@@ -0,0 +1,158 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import org.broadinstitute.dsde.workbench.fixture.BillingFixtures
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.service.{Orchestration, RestException}
+import org.scalatest.{FreeSpec, ParallelTestExecution}
+
+/**
+  * Created by rtitle on 4/26/18.
+  */
+class ClusterConcurrencySpec extends FreeSpec with LeonardoTestUtils with ParallelTestExecution with BillingFixtures {
+
+  // these tests just hit the Leo APIs; they don't interact with notebooks via selenium
+  "Leonardo cluster status transitions" - {
+
+    // (create -> wait -> delete -> wait) * 2
+    "should create, monitor, delete, recreate, and re-delete a cluster" in {
+      withCleanBillingProject(hermioneCreds) { projectName =>
+        Orchestration.billing.addUserToBillingProject(projectName, ronEmail, Orchestration.billing.BillingProjectRole.User)(hermioneAuthToken)
+        val project = GoogleProject(projectName)
+        implicit val token = ronAuthToken
+        val nameToReuse = randomClusterName
+
+        // create, monitor, delete once
+        withNewCluster(project, nameToReuse)(_ => ())
```

Comment:
It could, but then it would return `Cluster` instead of `Unit`. We don't use the return value anyway so that doesn't really matter. :)

I'm curious, is there a function like `identity` which instead of `A => A` is `A => Unit`? I can't find one.